15-components-lab_v16/src/01-timeconverter.jsx

```diff
@@ -8,5 +8,38 @@
 
 import React from 'react';
 
-export default React.createClass({
-});
+
+export default class TimeConverter extends React.Component {
+
+    constructor(props){
+        super(props);
+        this.state = { hours : 1, minutes: 60, seconds: 3600}
+    }
+
+    hourUpdate = (e) => {
+        const val = this.minutes.value;
+        this.setState(oldState => ({ hours: val/60, minutes : val, seconds: val * 60 }));
+    }
+
+    minUpdate = (e) => {
```

Comment:
The best practice in such cases is to use the event object your function already receives. That way you don't need to keep DOM reference.

```javascript
// e.target.value is the value user typed in the input field
const val = e.target.value;
this.setState(...);
```